### DISCUSSION TITLE
Help, the agent node has multiple network cards, and --flannel-iface does not seem to work

### DISCUSSION THREAD

**Discussion by Cairry** [k3s-io/k3s#10948] [Q&A]
**Environmental Info:**
K3s Version: v1.24.17-rc2+k3s1 (0d646e40)

Node(s) CPU architecture, OS, and Version: Linux ali-f-tair-vm-20-26 3.10.0-1160.114.2.el7.x86_64 #1 SMP Wed Mar 20 15:54:52 UTC 2024 x86_64 x86_64 x86_64 GNU/Linux

Cluster Configuration: 1 server, 1 agent

**Describe the bug:**
To put it simply, the node where the Pod is located is 1.1.1.1. Pinging 1.1.1.1 in the Pod does not work, but pinging any other node such as 1.1.1.2 works.

A new machine with a single network card was added, and the host machine to which the sleep-27 pod request belongs is OK. The GPU machine has multiple network cards, and the host machine to which the sleep pod request belongs is not connected.
![image](https://github.com/user-attachments/assets/51b98890-4fdf-44d5-a040-1c5ce0593b25)

**Comment by brandond** [comment]
> The GPU machine has multiple network cards, and the host machine to which the sleep pod request belongs is not connected.

I'm not sure what you mean with this statement. Are you saying that the pod is on a node that is disconnected from the cluster? Did you perhaps need to set --node-ip in addition to --flannel-iface? Are all of your nodes on the same network, such that they can reach each other at their internal IPs?
